server/src/main/java/org/opensearch/client/node/NodeClient.java

```diff
@@ -82,6 +93,21 @@ public void initialize(
         this.namedWriteableRegistry = namedWriteableRegistry;
     }
 
+    public void initialize(
+        Map<ActionType, TransportAction> actions,
+        ActionModule actionModule,
+        TransportService transportService,
+        ExtensionsManager extensionsManager,
+        Supplier<String> localNodeId,
+        NamedWriteableRegistry namedWriteableRegistry
+    ) {
+        initialize(actions, localNodeId, transportService.getRemoteClusterService(), namedWriteableRegistry);
+        this.extensionActions = actionModule.getExtensionActions();
```

Comment:
I don't like the abstraction of 'action' vs 'extensionAction' leaking into the NodeClient.  What do you think about encapsulating the extensionAction concepts fully inside of ActionModule?

Reply:
So I was able to refactor and get everything into a single registry.  Haven't tested whether it works but we'll see! :)

Reply:
🤞